src/odemis/acq/stitching/test/tiledacq_test.py

```diff
@@ -311,6 +311,119 @@ def test_whole_procedure(self):
+                        [0.00019889999999999998, 0.0001485167330631956, 4.765627e-06]]
+        focus_range = self.focus.getMetadata().get(model.MD_POS_ACTIVE_RANGE)
+        axis_range = self.focus.axes['z'].range
+
+        # Test 1 when z level is not provided
+        tiled_acq_task = TiledAcquisitionTask(self.fm_streams, self.stage,
+                                              area=area, overlap=overlap, future=model.InstantaneousFuture(),
+                                              focus_points=focus_points)
+        # TODO need the following metadata in mimas-sim and meteor-sim
+        # Only the enzel file is modified to run this testcase that uses enzel-sim
+        # TODO check the values of the metaadat POS_ACTIVE_RANGE in enzel-sim for focus role
+        tiled_acq_task._refocus()
+        # min with none values
+        zmin = max(min(axis_range), min(focus_range))
+        zmax = min(max(axis_range), max(focus_range))
+        # Test the min and max range of the zstack
+        self.assertGreaterEqual(tiled_acq_task._zlevels[0], zmin)
+        self.assertLessEqual(tiled_acq_task._zlevels[-1], zmax)
+
+        # Test 2 when z level is provided
+        zlevels = [focus_points[0][2], focus_points[1][2], focus_points[2][2]]
+
+        with self.assertRaises(NotImplementedError):
+            tiled_acq_task = TiledAcquisitionTask(self.fm_streams, self.stage,
+                                                  area=area, overlap=overlap, future=model.InstantaneousFuture(),
+                                                  zlevels=zlevels, focus_points=focus_points)
+
+    def test_get_z_on_focus_plane(self):
+        """Test the function that calculates the z position on the focus plane."""
+        area = (-0.001, -0.001, 0.001, 0.001)
+        overlap = 0.2
+        focus_points = [[-0.00019889999999999998, -0.0001485167330631956, 2.5185183e-05],
+                        [0.0, -0.0001485167330631956, -1.8148142e-05],
+                        [0.00019889999999999998, -0.0001485167330631956, -3.382517e-06],
+                        [-0.00019889999999999998, 0.0, 1.3804983e-05],
+                        [0.0, 0.0, 2.320607e-06], [0.00019889999999999998, 0.0, 3e-05],
+                        [-0.00019889999999999998, 0.0001485167330631956, -1.78125e-05],
+                        [0.0, 0.0001485167330631956, -8.124999e-06],
+                        [0.00019889999999999998, 0.0001485167330631956, 4.765627e-06]]
+        tiled_acq_task = TiledAcquisitionTask(self.fm_streams, self.stage,
+                                              area=area, overlap=overlap, future=model.InstantaneousFuture(),
+                                              focus_points=focus_points)
+        # Test a point that is on the plane
+        x = 0
+        y = 0
+        z = tiled_acq_task._get_z_on_focus_plane(x, y)
+        self.assertEqual(z, tiled_acq_task._focus_plane["gamma"])
+
+        # Test a point that is not on the plane
```

Comment:
why is this point not on the plane?

Reply:
deleting this test, does not work as intended. The preliminary idea was to check a 3d point outside the plane and make sure the z value from the plane calculation does not coincide with the z of 3d point.

Reply:
added better test